File: elearningit/lib/models/user_settings.dart
// models/user_settings.dart
class UserSettings {
  final String id;
  final String userId;
  final String theme; // 'light' or 'dark'

  UserSettings({required this.id, required this.userId, required this.theme});

  factory UserSettings.fromJson(Map<String, dynamic> json) {
    return UserSettings(
      id: json['_id'] ?? '',
      userId: json['userId'] ?? '',
      theme: json['theme'] ?? 'light',
    );
  }

  Map<String, dynamic> toJson() {
    return {'theme': theme};
  }

  UserSettings copyWith({String? theme}) {
    return UserSettings(id: id, userId: userId, theme: theme ?? this.theme);
  }

  static UserSettings defaultSettings() {
    return UserSettings(id: '', userId: '', theme: 'light');
  }
}

File: elearningit/backend/models/UserSettings.js
// models/UserSettings.js
const mongoose = require('mongoose');

const userSettingsSchema = new mongoose.Schema({
  userId: { 
    type: mongoose.Schema.Types.ObjectId, 
    ref: 'User', 
    required: true,
    unique: true 
  },
  theme: { 
    type: String, 
    enum: ['light', 'dark'], 
    default: 'light' 
  }
}, { 
  timestamps: true 
});

// Index for faster lookups
userSettingsSchema.index({ userId: 1 });

module.exports = mongoose.model('UserSettings', userSettingsSchema);
